compositor_pipeline/src/pipeline/input/rtp/depayloader.rs

```diff
@@ -204,9 +218,273 @@ impl RolloverState {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum AacDepayloadingError {
+    #[error("Packet too short")]
+    PacketTooShort,
+
+    #[error("Interleaving is not supported")]
+    InterleavingNotSupported,
+}
+
+impl AacDepayloaderMode {
+    fn size_len_in_bits(&self) -> usize {
+        match self {
+            AacDepayloaderMode::LowBitrate => 6,
+            AacDepayloaderMode::HighBitrate => 13,
+        }
+    }
+
+    fn index_len_in_bits(&self) -> usize {
+        match self {
+            AacDepayloaderMode::LowBitrate => 2,
+            AacDepayloaderMode::HighBitrate => 3,
+        }
+    }
+
+    fn header_len_in_bytes(&self) -> usize {
+        match self {
+            AacDepayloaderMode::LowBitrate => 1,
+            AacDepayloaderMode::HighBitrate => 2,
+        }
+    }
+}
+
+pub struct AacDepayloader {
+    mode: AacDepayloaderMode,
+    asc: Asc,
+    rollover_state: RolloverState,
+}
+
+/// MPEG-4 part 3, 1.6.3.4
+fn freq_id_to_freq(id: u8) -> Result<u32, AudioSpecificConfigParseError> {
+    match id {
+        0x0 => Ok(96000),
+        0x1 => Ok(88200),
+        0x2 => Ok(64000),
+        0x3 => Ok(48000),
+        0x4 => Ok(44100),
+        0x5 => Ok(32000),
+        0x6 => Ok(24000),
+        0x7 => Ok(22050),
+        0x8 => Ok(16000),
+        0x9 => Ok(12000),
+        0xa => Ok(11025),
+        0xb => Ok(8000),
+        0xc => Ok(7350),
+        _ => Err(AudioSpecificConfigParseError::IllegalValue),
+    }
+}
+
+/// MPEG-4 part 3, 4.5.1.1
+fn frame_length_flag_to_frame_length(flag: bool) -> u32 {
+    match flag {
+        true => 960,
+        false => 1024,
+    }
+}
+
+struct Asc {
+    _profile: u8,
+    frequency: u32,
+    _channel: u8,
+    frame_length: u32,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum AudioSpecificConfigParseError {
+    #[error("ASC is not long enough")]
+    TooShort,
+
+    #[error("Illegal value in ASC")]
+    IllegalValue,
+}
+
+// MPEG-4 part 3, sections 1.6.2.1 & 4.4.1
+fn parse_asc(asc: &[u8]) -> Result<Asc, AudioSpecificConfigParseError> {
```

Comment:
Extract this. This file is already huge

Reply:
this comment only loaded for me after i clicked merge, so I'll maybe do it in a subsequent PR

Reply:
sorry

Reply:
np